plugins/automation/spec/services/discourse_automation/destroy_spec.rb

```diff
@@ -17,15 +15,16 @@
     let(:params) { { automation_id: automation.id } }
     let(:dependencies) { { guardian: } }
 
-    it "logs the action" do
-      expect { result }.to change { UserHistory.count }.by(1)
-      expect(UserHistory.last.details).to eq(
-        "id: #{automation.id}\nname: #{automation.name}\nscript: #{automation.script}\ntrigger: #{automation.trigger}",
-      )
+    context "when user can't destroy the automation" do
+      fab!(:user) { Fabricate(:user) }
+
+      it { is_expected.to fail_a_policy(:can_destroy_automation) }
     end
 
-    it "destroys the automation" do
-      expect { result }.to change { DiscourseAutomation::Automation.count }.by(-1)
+    context "when data is invalid" do
```

Comment:
We were missing a test to make sure the contract is properly called inside the service.